### PR TITLE
chore: fix typecheck PVC and incorrect dispatch

### PR DESCRIPTION
**Pull request by cdrage**
chore: fix typecheck PVC and incorrect dispatch

### What does this PR do?

Fixes the typecheck, as well as the incorrect dispatch that was set.
Both should be set to false to match how services / ingress / etc is
hot-reloaded / refreshed in the interface.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

Fixes the below output:

```
Run yarn typecheck
yarn run v1.22.22
$ npm run typecheck:main && npm run typecheck:preload && npm run typecheck:ui && npm run typecheck:renderer && npm run typecheck:preload-dd-extension && npm run typecheck:preload-webview && npm run typecheck:extensions && npm run typecheck:extension-api

> podman-desktop@1.11.0-next typecheck:main
> tsc --noEmit -p packages/main/tsconfig.json

Error: packages/main/src/plugin/kubernetes-context-state.ts(668,14): error TS2554: Expected 5 arguments, but got 4.
Error: packages/main/src/plugin/kubernetes-context-state.ts(668,88): error TS7006: Parameter 'state' implicitly has an 'any' type.
Error: packages/main/src/plugin/kubernetes-context-state.ts(673,14): error TS2554: Expected 5 arguments, but got 4.
Error: packages/main/src/plugin/kubernetes-context-state.ts(673,88): error TS7006: Parameter 'state' implicitly has an 'any' type.
Error: packages/main/src/plugin/kubernetes-context-state.ts(675,13): error TS7006: Parameter 'o' implicitly has an 'any' type.
Error: packages/main/src/plugin/kubernetes-context-state.ts(681,14): error TS2554: Expected 5 arguments, but got 4.
Error: packages/main/src/plugin/kubernetes-context-state.ts(685,11): error TS7006: Parameter 'state' implicitly has an 'any' type.
Error: packages/main/src/plugin/kubernetes-context-state.ts(687,15): error TS7006: Parameter 'd' implicitly has an 'any' type.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 2.
```

### How to test this PR?

- [X] Tests are covering the bug fix or the new feature

Should pass CI now

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Review by axel7083 — APPROVED** [review]

File: packages/main/src/plugin/kubernetes-context-state.ts [hunk 1]
```diff
@@ -665,12 +665,12 @@ export class ContextsManager {
...
       connectionDelay: connectionDelay,
       onAdd: obj => {
-        this.setStateAndDispatch(context.name, true, { persistentvolumeclaims: true }, state =>
+        this.setStateAndDispatch(context.name, false, false, { persistentvolumeclaims: true }, state =>
           state.resources.persistentvolumeclaims.push(obj),
         );
       },
       onUpdate: obj => {
-        this.setStateAndDispatch(context.name, true, { persistentvolumeclaims: true }, state => {
+        this.setStateAndDispatch(context.name, false, false, { persistentvolumeclaims: true }, state => {
```
Comment (feloy) [inline packages/main/src/plugin/kubernetes-context-state.ts:673]:
```suggestion
        this.setStateAndDispatch(context.name, false, true, { persistentvolumeclaims: true }, state => {
```

**Review by feloy — APPROVED** [review]
LGTM

**Comment by cdrage** [comment]
@feloy I noticed in Ingress, Routes, Node and Service it is all `false, false`, why PVC and Deployment `false, true`?

Should we change the other ones too?

**Comment by feloy** [comment]
> @feloy I noticed in Ingress, Routes, Node and Service it is all `false, false`, why PVC and Deployment `false, true`?
> 
> Should we change the other ones too?

The first one should always be false here (it should be true only when will send the reachable checking state).

The second should be true for pods and deployments only ("primary" resources visible in the Kube contexts list), and false for the others ("secondary" resources).

Please ignore my first comment (that I deleted in the meantime)

**Comment by axel7083** [comment]
> > @feloy I noticed in Ingress, Routes, Node and Service it is all `false, false`, why PVC and Deployment `false, true`?
> > Should we change the other ones too?
> 
> The first one should always be false here (it should be true only when will send the reachable checking state).
> 
> The second should be true for pods and deployments only ("primary" resources visible in the Kube contexts list), and false for the others ("secondary" resources).
> 
> Please ignore my first comment (that I deleted in the meantime)

As this feel to be confusing

https://github.com/containers/podman-desktop/blob/2a31670e85e87a2289847720570b8245e26fb589/packages/main/src/plugin/kubernetes-context-state.ts#L998-L1002

Do you think @feloy you could add some documentation in `kubernetes-context-state.ts` for the functions (arguments, behaviour etc.), as this is a bit hard to understand when reading the code, and you were the last one working on it, would be very appreciate :)

**Comment by feloy** [comment]
> > > @feloy I noticed in Ingress, Routes, Node and Service it is all `false, false`, why PVC and Deployment `false, true`?
> > > Should we change the other ones too?
> > 
> > 
> > The first one should always be false here (it should be true only when will send the reachable checking state).
> > The second should be true for pods and deployments only ("primary" resources visible in the Kube contexts list), and false for the others ("secondary" resources).
> > Please ignore my first comment (that I deleted in the meantime)
> 
> As this feel to be confusing
> 
> https://github.com/containers/podman-desktop/blob/2a31670e85e87a2289847720570b8245e26fb589/packages/main/src/plugin/kubernetes-context-state.ts#L998-L1002
> 
> Do you think @feloy you could add some documentation in `kubernetes-context-state.ts` for the functions (arguments, behaviour etc.), as this is a bit hard to understand when reading the code, and you were the last one working on it, would be very appreciate :)

Yes, sure, I'll also change the parameters to pass a structure of options instead.

**Pull request merged**